sink/src/main/java/com/spredfast/kafka/connect/s3/sink/S3Writer.java

```diff
@@ -33,38 +27,21 @@
  */
 public class S3Writer {
   private static final Logger log = LoggerFactory.getLogger(S3SinkTask.class);
-  private static final TimeZone UTC = TimeZone.getTimeZone("UTC");
   private final ObjectReader reader = new ObjectMapper().readerFor(ChunksIndex.class);
-  private String keyPrefix;
-  private String bucket;
+  private final String keyPrefix;
+  private final String bucket;
   private final Layout.Builder layoutBuilder;
-  private AmazonS3 s3Client;
-  private TransferManager tm;
```

Comment:
Were there any advantages that the TransferManager provided us? It does exist for the AWS SDK for Java V2: https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/transfer-manager.html. I don't know offhand.

Reply:
I came to the conclusion that it makes no sense to use transfer manager here because we did not use any features it provides on top of the s3 client. Transfer manager is used when you need to upload/download big amount of objects, and you don't want to write your own pool of threads, traverse the file system and the bucket content, distribute and monitor progress and so on. In our case, we just need to upload 3 objects sequentially, so it makes more sense to just use the s3 client.

Another more practical reason why it's better to remove it is that in SDK V2 it requires an async s3 client. Again, it makes no sense to use an async s3 client only to mimic a synchronous s3 client. In the beginning I was trying to keep the transfer manager, but it was super hard because it required to rewrite all usages of s3 client in an async fashion without giving any benefits. 

In the end, I replaced the usage of the transfer manager to upload 2 objects with two `putObject` calls.